common/common.cpp

```diff
@@ -2516,6 +2527,37 @@ bool create_directory_with_parents(const std::string & path) {
 #endif // _WIN32
 }
 
+std::string get_cache_directory() {
+    std::string cache_directory = "";
+    if (getenv("LLAMA_CACHE")) {
+        cache_directory = std::getenv("LLAMA_CACHE");
+        if (cache_directory.back() != DIRECTORY_SEPARATOR) {
+            cache_directory += DIRECTORY_SEPARATOR;
+        }
+    } else {
+#ifdef __linux__
+        if (std::getenv("XDG_CACHE_HOME")) {
+            cache_directory = std::getenv("XDG_CACHE_HOME");
+        } else {
+            cache_directory = std::getenv("HOME") + std::string("/.cache/");
+        }
+#elif defined(__APPLE__)
+        cache_directory = std::getenv("HOME") + std::string("/Library/Caches/");
```

Comment:
i think on Mac os pretty much all apps also use `~/.cache` (so, same as linux)

Just my 2 cents

Reply:
I don't use macos very often so I'm not sure. Maybe @ggerganov knows more about this?

My assumption is that `~/.cache` widely used by applications ported from linux (maybe?), while `~/Library/Caches` is the conventional OS X way (according to [this documentation](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFileSystem/Articles/WhereToPutFiles.html)).

In either way, having model cached under user's $HOME is already very nice.

Reply:
I see apps caches in both places on my machine - I guess either way is fine. Don't have an opinion

Reply:
Let's keep `~/Library/Caches` then